### PR TITLE
feat(spotify-play): Add functionality to play songs on spotify

### PR DESCRIPTION
**Pull request by 320834**
Branched off of this: https://github.com/seungguini/vintage_turntable/pull/20.

- Play/Pause for spotify

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -85,9 +86,14 @@ const Scene = () => {
...
-    <Canvas shadows dpr={[1, 2]} camera={{ position: [0, 0, 10], fov: 50 }}>
-      <Scene />
-    </Canvas>
+    <>
+      <React.StrictMode>
+        <Canvas shadows dpr={[1, 2]} camera={{ position: [0, 0, 10], fov: 50 }}>
+          <Scene />
+        </Canvas>
+      </React.StrictMode>
+      <SpotifyPlayer></SpotifyPlayer>
```
Comment (seungguini) [inline frontend/src/App.tsx]: NIT this can be `<SpotifyPlayer />` right?

File: frontend/src/hooks/playbackHooks.ts [hunk 2]
```diff
@@ -1,30 +1,57 @@
...
   useEffect(() => {
     if (!isPlaying) {
-      song.pause();
-      TONE_ARM_SOUND_EFFECT.play();
-      VINYL_SOUND_EFFECT.pause();
+      if(spotifyPlayer) {
+        pauseSpotify(spotifyPlayer);
+      } else {
+        song.pause();
+        TONE_ARM_SOUND_EFFECT.play();
```
Comment (seungguini) [inline frontend/src/hooks/playbackHooks.ts]: Add vinyl effects for the Spotify section as well.

File: frontend/src/hooks/playbackHooks.ts [hunk 3]
```diff
@@ -1,30 +1,57 @@
...
   useEffect(() => {
     if (isPlaying && toneArmFinished) {
-      console.log("Play button hit + tone arm moved");
-      TONE_ARM_SOUND_EFFECT.play();
-      VINYL_SOUND_EFFECT.play();
-      song.play();
+      if(spotifyPlayer) {
+        playSpotify();
+      } else {
+        console.log("Play button hit + tone arm moved");
```
Comment (seungguini) [inline frontend/src/hooks/playbackHooks.ts]: same here, let's add vinyl effect for the spotify section too.

**Review by seungguini — CHANGES_REQUESTED** [review]

File: frontend/src/App.tsx (hunk 1 above)
Comment (320834) [reply frontend/src/App.tsx]: yep

**Review by seungguini — APPROVED** [review]
LGTM. Great work!

**Pull request merged**